javalib-ext-dummies/src/main/scala/java/time/Instant.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.time
+
+final class Instant(private val epochSecond: Long, private val nano: Int) {
+  def getEpochSecond(): Long = epochSecond
+
+  def getNano(): Int = nano
+
+  def toEpochMilli(): Long = {
+    if (epochSecond == -9223372036854776L) {
+      /* Special case: epochSecond * 1000L would overflow, but the addition
+       * of the nanos might save the day. So we transfer one unit of the
+       * seconds into the contribution of the nanos.
+       */
+      Math.addExact(-9223372036854775000L, (nano / 1000000) - 1000)
+    } else {
+      Math.addExact(Math.multiplyExact(epochSecond, 1000L), nano / 1000000)
+    }
+  }
+
+  def isAfter(otherInstant: Instant): Boolean = {
+    this.epochSecond > otherInstant.epochSecond || {
+      this.epochSecond == otherInstant.epochSecond &&
+      this.nano > otherInstant.nano
+    }
+  }
+
+  def isBefore(otherInstant: Instant): Boolean = {
+    this.epochSecond < otherInstant.epochSecond || {
+      this.epochSecond == otherInstant.epochSecond &&
+      this.nano < otherInstant.nano
+    }
+  }
+
+  override def equals(that: Any): Boolean = that match {
+    case that: Instant =>
+      this.epochSecond == that.epochSecond &&
+      this.nano == that.nano
+    case _ =>
+      false
+  }
+
+  override def hashCode(): Int =
+    java.lang.Long.hashCode(epochSecond) ^ java.lang.Integer.hashCode(nano)
+
+  // non compliant, for debugging purposes only
+  override def toString(): String =
+    "Instant(" + epochSecond + ", " + nano + ")"
+}
+
+object Instant {
+  final val EPOCH: Instant = new Instant(0L, 0)
+  final val MIN: Instant = new Instant(-31557014167219200L, 0)
+  final val MAX: Instant = new Instant(31556889864403199L, 999999999)
+
+  private def checkAndCreate(epochSecond: Long, nano: Int): Instant = {
+    val instant = new Instant(epochSecond, nano)
+    if (instant.isBefore(MIN) || instant.isAfter(MAX))
+      throw new DateTimeException("Instant exceeds minimum or maximum instant")
+    instant
+  }
```

Comment:
Do we need this check? It seems our implementation doesn't use it (neither do we test it).

Reply:
Technically, we don't need it, but if we don't do the check, we could have another implementation that looks like it works, although it would transitively throw a `DateTimeException` when using a compliant `Instant`.

Should I test the behavior instead?

Reply:
> Technically, we don't need it, but if we don't do the check, we could have another implementation that looks like it works, although it would transitively throw a DateTimeException when using a compliant Instant.

I'm not sure I follow. You mean another plausible implementation of `j.u.Date`?

Reply:
Another plausible implementation of something (not specifically `j.u.Date`) that we might implement in the future.

I just find it dangerous *not to throw* something that should be thrown. Because we will never know in the future that we're relying on something that would in fact break. It is more dangerous than not implementing all the positive behaviors, because if we start relying on those, we would immediately notice that something's wrong.

Reply:
Ah, fair enough. Yes. Then maybe we should just add an additional test case checking that it throws.

Reply:
Done.